panos_upgrade_assurance/snapshot_compare.py

```diff
@@ -129,30 +129,38 @@ def compare_snapshots(
         return result
 
     @staticmethod
-    def key_checker(left_dict: dict, right_dict: dict, key: str) -> None:
-        """The static method to check if a key is available in both dictionaries.
+    def key_checker(left_dict: dict, right_dict: dict, key: Union[str, set, list]) -> None:
+        """The static method to check if a key(string) or list/set of keys is available in both dictionaries.
 
-        This method looks for a given key in two dictionaries. Its main purpose is to assure that when comparing a key-value pair from two dictionaries, it actually exists in both.
+        This method looks for a given key or list/set of keys in two dictionaries. Its main purpose is to assure that when comparing a key-value pair from two dictionaries, it actually exists in both.
 
         # Parameters
 
         left_dict (dict): 1st dictionary to verify.
         right_dict (dict): 2nd dictionary to verify.
-        key (str): Key name to check.
+        key (str, set, list): Key name or set/list of keys to check.
 
         # Raises
 
         MissingKeyException: when key is not available in at least one snapshot.
 
         """
-        left_snap_missing_key = False if key in left_dict else True
-        right_snap_missing_key = False if key in right_dict else True
+
+        if isinstance(key, str):
+            key_set = [key]
+        elif isinstance(key, (set, list)):
+            key_set = key
+        else:
+            raise WrongDataTypeException(f'The key variable is a {type(key)} but should be either str, set or list')
+
+        left_snap_missing_key = False if set(key_set).issubset(left_dict.keys()) else True
+        right_snap_missing_key = False if set(key_set).issubset(right_dict.keys()) else True
```

Comment:
following suggestion from line 149, this should be something like this:

```suggestion
        left_snap_missing_key = False if key_set.issubset(left_dict.keys()) else True
        right_snap_missing_key = False if key_set.issubset(right_dict.keys()) else True
```